frontend/src/routes/Infrastructure/Clusters/ManagedClusters/ImportCluster/ImportCluster.tsx

```diff
@@ -97,6 +188,88 @@ export function ImportClusterPageContent() {
+    const [importResources, setImportResources] = useState<any | undefined>([])
+
+    useEffect(() => {
+        /* istanbul ignore next */
+        const clusterLabels: Record<string, string> = {
+            cloud: 'auto-detect',
+            vendor: 'auto-detect',
+            name: clusterName,
+            ...additionalLabels,
+        }
+        if (managedClusterSet) {
+            clusterLabels[managedClusterSetLabel] = managedClusterSet
+        }
+        let clusterAnnotations: Record<string, string> = {}
+        if (discovered) {
+            clusterAnnotations = {
+                'open-cluster-management/created-via': 'discovery',
+            }
+        }
+        const resources = []
+        resources.push({
+            apiVersion: ManagedClusterApiVersion,
+            kind: ManagedClusterKind,
+            metadata: {
+                name: clusterName,
+                labels: clusterLabels,
+                annotations: clusterAnnotations,
+            },
+            spec: { hubAcceptsClient: true },
+        })
+
+        switch (importMode) {
+            case ImportMode.kubeconfig:
+                resources.push({
+                    apiVersion: SecretApiVersion,
+                    kind: SecretKind,
+                    metadata: {
+                        name: 'auto-import-secret',
+                        namespace: clusterName,
+                    },
+                    stringData: {
+                        autoImportRetry: '2',
+                        kubeconfig: kubeConfig,
+                    },
+                    type: 'Opaque',
+                })
+                break
+            case ImportMode.token:
+                resources.push({
+                    apiVersion: SecretApiVersion,
+                    kind: SecretKind,
+                    metadata: {
+                        name: 'auto-import-secret',
+                        namespace: clusterName,
+                    },
+                    stringData: {
+                        autoImportRetry: '2',
+                        token: token,
+                        server: server,
+                    },
+                    type: 'Opaque',
+                })
+        }
+        resources.push({
+            apiVersion: KlusterletAddonConfigApiVersion,
+            kind: KlusterletAddonConfigKind,
+            metadata: { name: clusterName, namespace: clusterName },
+            spec: {
+                clusterName: clusterName,
+                clusterNamespace: clusterName,
+                clusterLabels: { ...clusterLabels },
+                applicationManager: { enabled: true, argocdCluster: false },
+                policyController: { enabled: true },
+                searchCollector: { enabled: true },
+                certPolicyController: { enabled: true },
+                iamPolicyController: { enabled: true },
+                version: '2.2.0',
```

Comment:
I believe this is deprecated and can be removed here and elsewhere where the version is specified in the KlusterletAddonConfig. See https://github.com/open-cluster-management/klusterlet-addon-controller/blob/main/pkg/apis/agent/v1/klusterletaddonconfig_types.go#L20

Reply:
deleted--ha--couldn't really see the problem until we exposed the yaml :)